src/library/trackset/setlogfeature.cpp

```diff
@@ -105,6 +108,46 @@ SetlogFeature::~SetlogFeature() {
     deleteAllUnlockedPlaylistsWithFewerTracks();
 }
 
+
+void SetlogFeature::addPlaylistIfCmdArg() {
+    const QString playlistFile = CmdlineArgs::Instance().getPlaylistFilePath();
+    if(!playlistFile.isNull()) {
+        int lastPlaylistId = kInvalidPlaylistId;
+
+        const QFileInfo fileInfo(playlistFile);
+        // Get a valid name
+        const QString baseName = fileInfo.baseName();
+        QString name;
+
+        bool validNameGiven = false;
+        int i = 0;
+        while (!validNameGiven) {
+            name = baseName;
+            if (i != 0) {
+                name += QString::number(i);
+            }
+
+            // Check name
+            int existingId = m_playlistDao.getPlaylistIdFromName(name);
+
+            validNameGiven = (existingId == kInvalidPlaylistId);
+            ++i;
+        }
```

Comment:
Btw I opened #12246 to simplify / fix the original implementation in `BasePlaylistFeature::slotCreateImportPlaylist`